src/utils/array.ts

```diff
@@ -110,3 +110,21 @@ export function depth(arr: unknown, base = 0) {
 	}
 	return depth(arr[0], base + 1);
 }
+
+/**
+ * Find the longest array from a list of arrays
+ */
+export function findLongest<T>(arr: T[]): T {
+	let longest = arr[0];
+	let max = -1;
+	for (const item of arr) {
+		if (Array.isArray(item) && item.length > max) {
+			max = item.length;
+			longest = item;
+		}
+		if (!Array.isArray(item) && max < 0) {
+			longest = item;
+		}
+	}
+	return longest;
+}
```

Comment:
What do you think about : 

```ts
export function findLongest<T extends unknown[]>(arr: T[]): T {
  return arr.reduce((longest, item) => 
    item.length > longest.length ? item : longest, 
    arr[0]
  );
}
```

Reply:
Il faut alors ajouter une condition car je n'ai pas de garantit qu'item soit un tableau :(

Reply:
Tu pourrais, en appelant la function qu'avec des tableaux ? 
 
```
findLongest(this.shapeFrom.map((v) => this.dictionary?.get(v)?.getValue())
 .filter(Array.isArray)
 )
 ```

Reply:
Using the filter function requires traversing the array twice, which is not very efficient. A better solution is to use the reduce function, although it can make the code more complex.